common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftFamilyOAuth2TokenCache.java

```diff
@@ -145,8 +149,15 @@ public ICacheRecord loadByFamilyId(@Nullable final String clientId,
                             && accountRecord.getHomeAccountId().equals(atRecord.getHomeAccountId())
                             && accountRecord.getRealm().equals(atRecord.getRealm())
                             && targetsIntersect(target, atRecord.getTarget())) {
-                        atRecordToReturn = atRecord;
-                        break;
+                        if (CredentialType.AccessToken.name().equalsIgnoreCase(atRecord.getCredentialType())
+                                && BearerAuthenticationSchemeInternal.SCHEME_BEARER.equalsIgnoreCase(authenticationScheme.getName())) {
```

Comment:
atRecord.getCredentialType() could be null for tokens cached before this change.

Reply:
Sync'd offline. We're good here 👍